include/deal.II/base/parameter_handler.h

```diff
@@ -1156,6 +1156,110 @@ namespace Patterns
+ * somewhere outside the ParameterHandler object. (Exactly when the
+ * action is called is described in the documentation of the
+ * add_action() function.) Of course, in C++ one doesn't usually pass
+ * around the address of a function, but an action can be a function-like
+ * object (taking a string as argument) that results from calling
+ * @p std::bind, or more conveniently, it can be a
+ * <a href="http://en.cppreference.com/w/cpp/language/lambda">lambda
+ * function</a> that has the form
+ * @code
+ *   [] (const std::string &value) { ... do something with the value ... }
+ * @endcode
+ * and that is attached to a specific parameter.
+ *
+ * A typical example of such an action would be as follows: let's assume
+ * that you have a program that declares a parameter for the number
+ * of iterations it is going to run, say
+ * @code
+ *   class MyAlgorithm
+ *   {
+ *      public:
+ *        void run ();
+ *      private:
+ *        unsigned int n_iterations;
+ *   };
+ * @endcode
+ * then one could obtain this parameter from a parameter file using a
+ * code snippet in @p run() as follows:
+ * @code
+ *   void MyAlgorithm::run ()
+ *   {
+ *     ParameterHandler prm;
+ *     prm.declare_entry ("Number of iterations",  // name of parameter
+ *                        "10",                    // default value
+ *                        Patterns::Integer(1,100),// allowed values: 1...100
+ *                        "The number of ...");    // some documentation, to be completed
+ *
+ *     // next read the parameter from an input file...
+ *     prm.parse_input ("my_algorithm.prm");
+ *
+ *     // ...and finally get the value for use in the program:
+ *     n_iterations = prm.get_integer ("Number of iterations");
+ *
+ *     ... actual code doing something useful follows here...
+ * @endcode
+ *
+ * This two-step process -- first declaring the parameter, and later reading
+ * it -- is a bit cumbersome because one has to first declare <i>all</i>
+ * parameters and at a later time retrieve them from the ParameterHandler
+ * object. In large programs, these two things also often happen in
+ * different functions.
+ *
+ * To avoid this, it would be nice if we could put both the declaration
+ * and the retrieval into the same place. This can be done via actions,
+ * and the function would then look like this:
+ * @code
+ *   void MyAlgorithm::run ()
+ *   {
+ *     ParameterHandler prm;
+ *     prm.declare_entry ("Number of iterations",  // name of parameter
+ *                        "10",                    // default value
+ *                        Patterns::Integer(1,100),// allowed values: 1...100
+ *                        "The number of ...");    // some documentation, to be completed
+ *     prm.add_action ("Number of iterations",
+ *                     [&](const std::string &value) {
+ *                       this->n_iterations = Utilities::string_to_int(value);
+ *                     });
```

Comment:
Somehow, this is still a two-step process. But I like it, since it allows for more actions to be added to the same parameter.